### PR TITLE
Rerun `transcriptomic-dimension-reduction` module

### PR DESCRIPTION
**Pull request by cbethell**
- rerun module with v14 data files 
- add `generate-multipanel-plot-functions.R` script in `util`
- implement use of custom function in `generate-multipanel-plot.R` 
- propogate changes to `03-multipanel-plots.sh`

### Purpose/implementation Section

The purpose of this PR is to rerun the `transcriptomic-dimension-reduction` module using the v14 data release files. 

This PR also adds a function script that can be used in the `generate-multipanel-plot.R` script of this module, in the `09-HGG-with-braf-clustering.Rmd` notebook in the `analyses/molecular-subtyping-HGG` module (open PR #534), and in any other future instances where the generation of a multipanel plot is needed.

#### What scientific question is your analysis addressing?

None, this PR is just updating the module and refactoring a script.

#### What GitHub issue does your pull request address?

This PR does not address one issue in particular, it instead stems from the [comment made on open PR #534](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/534#discussion_r379169175).

#### Which areas should receive a particularly close look?

The`generate-multipanel-plot-functions.R` script in this module's `util` directory and the `generate-multipanel-plot.R` script in this module's `scripts` directory should both receive a close look as these contain the new changes.

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes, this is ready for review.

#### What types of results are included (e.g., table, figure)?

No additional results have been added to this module.

#### Reproducibility Checklist

This PR is an update to the module so this all was done previously.
- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/transcriptomic-dimension-reduction/scripts/generate-multipanel-plot.R [hunk 1]
```diff
@@ -40,6 +46,11 @@ if (!dir.exists(plot_directory)) {
   dir.create(plot_directory, recursive = TRUE)
 }
 
+#### Source custom function ---------------------------------------------------
```
Comment (jaclyn-taroni) [inline analyses/transcriptomic-dimension-reduction/scripts/generate-multipanel-plot.R]: Why not hard code the script that needs to be sourced? See another example from this module.

https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/46cf6ccb119312ccae6122ac94c51710df01f6da/analyses/transcriptomic-dimension-reduction/scripts/run-dimension-reduction.R#L56

Do you see any reason why the input would change?

Comment (cbethell) [reply analyses/transcriptomic-dimension-reduction/scripts/generate-multipanel-plot.R]: Good question. I do not anticipate the input changing so I'll go ahead and hard code this.

**Comment by cbethell** [comment]
In the last commit, I was able to retain the values that were "missing" from the `molecular_subtype` variable in the aligned output TSV files, but was unable to determine the source of variation in the dimension reduction scores. 

I confirmed that the values in the expression files (from v13 and v14 releases) do match, and the seed remained the same (hence, my confusion).

**Comment by jaclyn-taroni** [comment]
I think the only method with notable changes is t-SNE. However, I've been poking around locally using the project Docker image and running the same thing multiple times does not result in different values, which suggests to me that the `set.seed` is being respected/is in the correct spot/etc. The first time I ran it locally (https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/542/commits/9e58fe2aa613900cf16503f70d638b7e4790b516) I _did_ get different values than what was in this branch (minor changes in PCA, differences in t-SNE, UMAP was the same) so I am wondering if it comes down to some difference in environment.

**Pull request merged**